jquery.canvaswrapper.js

```diff
@@ -528,6 +530,220 @@
+            // displays have that many more pixels in the same advertised space.
+    
+            // Resizing should reset the state (excanvas seems to be buggy though)
+    
+            if (this.width !== width) {
+                scale.x = 1;
+                element.width = width * pixelRatio;
+                element.style.width = width + 'px';
+                this.width = width;
+                shouldresize = true; 
+            }
+    
+            if (this.height !== height) {
+                scale.y = 1;
+                element.height = height * pixelRatio;
+                element.style.height = height + 'px';
+                this.height = height;
+                shouldresize = true;
+            }
+
+            scale.z = 1;
+
+            // Scale the coordinate space to match the display density; so even though we
+            // may have twice as many pixels, we still want lines and other drawing to
+            // appear at the same size; the extra pixels will just make them crisper.
+
+            if (renderer && shouldresize) {
+                renderer.clear();
+                mainscene.userData.rendererSize.width = width;
+                mainscene.userData.rendererSize.height = height;
+
+                renderer.setClearColor(0xff0000, 0);
+                renderer.setSize(width, height, false);
+                renderer.setPixelRatio(this.pixelRatio);
+
+                camera.aspect = width/height;
+                camera.updateProjectionMatrix();
+
+                this.cameraSight.x = width / 2;
+                this.cameraSight.y = height / 2;
+                this.cameraSight.z = 1000;
+                
+                camera.position.set(width / 2, height / 2, 0);
+                camera.lookAt(this.cameraSight);
+                camera.updateMatrixWorld();
+            }
+    };
+
+    WebGlCanvas.prototype.clear = function() {
+        var renderer = this.renderer,
+            scenes = this.scenes,
+            mainscene = this.mainscene;
+
+        if (renderer) {
+            // Clear the canvas
+            renderer.setClearColor(0x0f0f0f, 0);
+            renderer.setScissorTest(false);
+            renderer.clear();
+
+            while (mainscene.children.length > 0) {
+                mainscene.remove(mainscene.children[0]);
+            }
+
+            renderer.setClearColor(0xff0000, 0);
+            renderer.setScissorTest(true);
+        }
+    };
+    /**
+     * Render the mainscene to 
+     * 
+     */
+    var defaultPlotOffset = {
+        left: 0,
+        right: 0,
+        top: 0,
+        bottom: 0
+    }
+    WebGlCanvas.prototype.render = function() {
+        var renderer = this.renderer,
+            camera = this.camera,
+            mainscene = this.mainscene,
+            plotOffset = mainscene.userData.plotOffset || defaultPlotOffset,
+            scale = this.scale,
+            rendererSize;
+
+        if (renderer) {
```

Comment:
Set the camera position, camera view point, renderer size and viewport before rendering to screen